### PR TITLE
fix: chainspec ttd check

### PR DESCRIPTION
**Pull request by rkrasiuk**
According to [EIP-3675](https://eips.ethereum.org/EIPS/eip-3675), the ttd fork is activated on the first block **after** the block that passed the terminal total difficulty threshold.

### CODE REVIEW

File: crates/primitives/src/chain/spec.rs [hunk 1]
```diff
@@ -472,12 +472,20 @@ impl ForkCondition {
...
-    pub fn active_at_ttd(&self, ttd: U256) -> bool {
+    pub fn active_at_ttd(&self, ttd: U256, difficulty: U256) -> bool {
         if let ForkCondition::TTD { total_difficulty, .. } = self {
-            ttd >= *total_difficulty
+            if ttd >= difficulty {
+                (ttd - difficulty) >= *total_difficulty
+            } else {
+                // invalid difficulty
+                false
+            }
```
Comment (rkrasiuk) [inline crates/primitives/src/chain/spec.rs:488]: i don't mind panicking here tbh

File: crates/net/network/src/config.rs [hunk 2]
```diff
@@ -307,8 +307,9 @@ impl NetworkConfigBuilder {
         let head = head.unwrap_or(Head {
             hash: chain_spec.genesis_hash(),
             number: 0,
-            total_difficulty: U256::ZERO,
             timestamp: 0,
+            difficulty: chain_spec.genesis.difficulty,
+            total_difficulty: U256::ZERO,
```
Comment (rkrasiuk) [inline crates/net/network/src/config.rs:312]: @onbjerg shouldn't this be set to `chain_spec.genesis.difficulty` as well?

File: crates/primitives/src/chain/spec.rs [hunk 3]
```diff
@@ -472,12 +472,15 @@ impl ForkCondition {
         }
     }
 
-    /// Checks whether the fork condition is satisfied at the given total difficulty.
+    /// Checks whether the fork condition is satisfied at the given total difficulty and difficulty
+    /// of a current block.
+    ///
+    /// The fork is considered active if the previous total difficulty is above the threshold.
```
Comment (gakonst) [inline crates/primitives/src/chain/spec.rs:478]:
```suggestion
    /// The fork is considered active if the _previous_ total difficulty is above the threshold.
    /// To achieve that, we subtract the passed `difficulty` from the current block's total difficulty,
    /// and check if it's above the Fork Condition's total difficulty (here: 58_750_000_000_000_000_000_000)
```

File: crates/net/network/src/config.rs (hunk 2 above)
Comment (gakonst) [inline crates/net/network/src/config.rs:312]: I think this ultimately won't really matter as we'll override it inside of the session? cc @mattsse 

https://github.com/paradigmxyz/reth/blob/4b0acce9577f6e518d1d7f192536c6ada9a07e95/crates/net/network/src/session/mod.rs#L182-L187

```suggestion
            total_difficulty: chain_spec.genesis.difficulty,
```

Comment (rkrasiuk) [reply crates/net/network/src/config.rs:312]: not a blocker for this PR, i will extract it to an issue

**Review by gakonst — APPROVED** [review]

**Review by gakonst — APPROVED** [review]

**Pull request merged**